File: core/descriptors/src/org/jetbrains/kotlin/descriptors/annotations/AnnotationArgumentVisitor.java
/*
 * Copyright 2010-2015 JetBrains s.r.o.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.jetbrains.kotlin.descriptors.annotations;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.kotlin.resolve.constants.*;
import org.jetbrains.kotlin.resolve.constants.StringValue;

public interface AnnotationArgumentVisitor<R, D> {
    R visitLongValue(@NotNull LongValue konstue, D data);

    R visitIntValue(IntValue konstue, D data);

    R visitErrorValue(ErrorValue konstue, D data);

    R visitShortValue(ShortValue konstue, D data);

    R visitByteValue(ByteValue konstue, D data);

    R visitDoubleValue(DoubleValue konstue, D data);

    R visitFloatValue(FloatValue konstue, D data);

    R visitBooleanValue(BooleanValue konstue, D data);

    R visitCharValue(CharValue konstue, D data);

    R visitStringValue(StringValue konstue, D data);

    R visitNullValue(NullValue konstue, D data);
    
    R visitEnumValue(EnumValue konstue, D data);
    
    R visitArrayValue(ArrayValue konstue, D data);

    R visitAnnotationValue(AnnotationValue konstue, D data);

    R visitKClassValue(KClassValue konstue, D data);

    R visitUByteValue(UByteValue konstue, D data);

    R visitUShortValue(UShortValue konstue, D data);

    R visitUIntValue(UIntValue konstue, D data);

    R visitULongValue(ULongValue konstue, D data);
}

File: core/compiler.common/src/org/jetbrains/kotlin/constant/AnnotationArgumentVisitor.kt
/*
 * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
 * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
 */

package org.jetbrains.kotlin.constant

abstract class AnnotationArgumentVisitor<R, D> {
    abstract fun visitLongValue(konstue: LongValue, data: D): R
    abstract fun visitIntValue(konstue: IntValue, data: D): R
    abstract fun visitErrorValue(konstue: ErrorValue, data: D): R
    abstract fun visitShortValue(konstue: ShortValue, data: D): R
    abstract fun visitByteValue(konstue: ByteValue, data: D): R
    abstract fun visitDoubleValue(konstue: DoubleValue, data: D): R
    abstract fun visitFloatValue(konstue: FloatValue, data: D): R
    abstract fun visitBooleanValue(konstue: BooleanValue, data: D): R
    abstract fun visitCharValue(konstue: CharValue, data: D): R
    abstract fun visitStringValue(konstue: StringValue, data: D): R
    abstract fun visitNullValue(konstue: NullValue, data: D): R
    abstract fun visitEnumValue(konstue: EnumValue, data: D): R
    abstract fun visitArrayValue(konstue: ArrayValue, data: D): R
    abstract fun visitAnnotationValue(konstue: AnnotationValue, data: D): R
    abstract fun visitKClassValue(konstue: KClassValue, data: D): R
    abstract fun visitUByteValue(konstue: UByteValue, data: D): R
    abstract fun visitUShortValue(konstue: UShortValue, data: D): R
    abstract fun visitUIntValue(konstue: UIntValue, data: D): R
    abstract fun visitULongValue(konstue: ULongValue, data: D): R
}
